### ISSUE TITLE
Draggable cannot choose to use root Overlay

### ISSUE THREAD

**Issue by monkeyswarm** [flutter/flutter#67338]
## Steps to Reproduce

Create an app with nested [Navigator]s. The non-root Navigator does not cover the full screen. Add within the non-root [Navigator], a [Draggable] with a visible feedback. Drag it.

**Expected results:** Draggable feedback is shown across the whole screen

**Actual results:** Draggable feedback is only shown within the bounds of the non-root [Navigator].

[Draggable] calls `Overlay.of(context)` to get the closest [Overlay]. Need to add a parameter to let user choose to use the root [Overlay] via `Overlay.of(context, rootOverlay:true)`

**Comment by TahaTesser** [comment]
Hi @monkeyswarm 
Can you please provide your `flutter doctor -v` your `flutter run --verbose` and a complete reproducible minimal code sample
Thank you

**Comment by monkeyswarm** [comment]
Flutter doctor output and code sample below. "flutter run --verbose" reveals private information.
I've already submitted a PR at https://github.com/flutter/flutter/pull/67375.

======
[✓] Flutter (Channel DraggableRootOverlay, 1.23.0-8.0.pre.143, on Mac OS X 10.15.6 19G73 x86_64, locale en-US)
    • Flutter version 1.23.0-8.0.pre.143 at /Users/diglesia/Source/flutter
    • Framework revision 6ae4e7861c (11 hours ago), 2020-10-05 20:57:08 -0700
    • Engine revision a8d7d97bef
    • Dart version 2.11.0 (build 2.11.0-185.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/diglesia/Library/Android/sdk
    • Platform android-stable, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio with Blaze.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode_11.7.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    ✗ CocoaPods installed but not working.
        You appear to have CocoaPods installed but it is not working.
        This can happen if the version of Ruby that CocoaPods was installed with is different from the one being used to invoke it.
        This can usually be fixed by re-installing CocoaPods. For more info, see https://github.com/flutter/flutter/issues/14293.
      To re-install CocoaPods, run:
        sudo gem install cocoapods

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio with Blaze.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Community Edition (version 2019.3.5)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547

[✓] Connected device (4 available)
    • SM T860 (mobile) • R52MA0M1B7Z • android-arm64  • Android 10 (API 29)
    • macOS (desktop)  • macos       • darwin-x64     • Mac OS X 10.15.6 19G73 x86_64
    • Web Server (web) • web-server  • web-javascript • Flutter Tools
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 85.0.4183.121
=====
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Column(
        children: <Widget>[
          Container(
              height: 200.0,
              child: Navigator(onGenerateRoute: (RouteSettings settings) {
                if (settings.name == '/') {
                  return MaterialPageRoute(
                      settings: settings,
                      builder: (context) => Container(
                          height: 300.0,
                          child: Center(
                            child: const Draggable<int>(
                              data: 1,
                              child: Text('Source'),
                              feedback: Text('Dragging'),
                            ),
                          )));
                }
                throw UnsupportedError('Unsupported route: $settings');
              })),
          DragTarget<int>(
            builder:
                (BuildContext context, List<int> data, List<dynamic> rejects) {
              return Container(
                  height: 300.0, child: Center(child: const Text('Target 1')));
            },
          ),
        ],
      ),
    );
  }
}
========
Result when dragging the [Draggable] towards the [DragTarget]
![screenshot-1601995961320](https://user-images.githubusercontent.com/1235083/95221898-c7a23d80-07ac-11eb-857e-8b8c7a33ec6e.jpg)

**Comment by TahaTesser** [comment]
### Preview
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/48603081/95298711-25608380-089a-11eb-82ce-1f993a8acb44.gif)


<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Column(
          children: <Widget>[
            Container(
                height: 200.0,
                child: Navigator(onGenerateRoute: (RouteSettings settings) {
                  if (settings.name == '/') {
                    return MaterialPageRoute(
                        settings: settings,
                        builder: (context) => Container(
                            height: 300.0,
                            child: Center(
                              child: const Draggable<int>(
                                data: 1,
                                child: Text('Source'),
                                feedback: Text('Dragging'),
                              ),
                            )));
                  }
                  throw UnsupportedError('Unsupported route: $settings');
                })),
            DragTarget<int>(
              builder: (BuildContext context, List<int> data,
                  List<dynamic> rejects) {
                return Container(
                    height: 300.0,
                    child: Center(child: const Text('Target 1')));
              },
            ),
          ],
        ),
      ),
    );
  }
}

```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 1.23.0-8.0.pre.185, on Mac OS X 10.15.7 19H2
    x86_64, locale en-GB)
    • Flutter version 1.23.0-8.0.pre.185 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision 6620f831e9 (6 hours ago), 2020-10-06 18:15:15 -0700
    • Engine revision 7e6191de07
    • Dart version 2.11.0 (build 2.11.0-186.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0.rc.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.0

[✓] Connected device (3 available)
    • macOS (desktop)  • macos      • darwin-x64     • Mac OS X 10.15.7 19H2
      x86_64
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome
      85.0.4183.121

• No issues found!
```

</details>

**Comment by goderbauer** [comment]
Closing as this should be fixed by #67375.
